static/js/rendered_markdown.js

```diff
@@ -242,4 +242,17 @@ export const update_elements = ($content) => {
             return ":" + text + ":";
         });
     }
+
+    // We need to add the .fa.fa-play class to video previews so that our CSS
+    // can show the play icon on hover, it would be better for this to be done
+    // on the backend, but that would not support legacy rendered markdown, so
+    // we just do it here.
```

Comment:
No, doing it on the backend would not be better.

`fa-play` is clearly a frontend-specific detail. We should endeavor to (a) keep frontend-specific details out of the backend and (b) keep the backend markup stable, so that we retain the flexibility we need to improve the frontend and support multiple frontends, without the frontend having to worry about all of the different ways previous versions of the backend attempted to accommodate previous versions of the frontend, for the rest of eternity.

We have thus far failed catastrophically at this endeavor, but we should avoid making the problem worse.

Reply:
Right. I'm remembering the AST over wire conversation on CZO.

@andersk do you think there would be any benefits to separating the logic that is used to "live update" eg
```
    $content.find("a.stream").each(function () {
        const stream_id = Number.parseInt($(this).attr("data-stream-id"), 10);
        if (stream_id && !$(this).find(".highlight").length) {
            // Display the current name for stream if it is not
            // being displayed in search highlight.
            const stream_name = stream_data.maybe_get_stream_name(stream_id);
            if (stream_name !== undefined) {
                // If the stream has been deleted,
                // stream_data.maybe_get_stream_name might return
                // undefined.  Otherwise, display the current stream name.
                $(this).text("#" + stream_name);
            }
        }
    });
```
and the logic that is used to "translate" whatever is sent from the backend to this frontend implementation?

Reply:
I think there's no particularly strong reason to separate those, but if we did, the way I'd expect to do that would be to extract the live-update parts to a function that's called both here and when a stream name is edited. (I think we have independent logic for those cases).